src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -692,8 +726,22 @@ def __call__(
 
         # 7. Denoising loop
         num_warmup_steps = len(timesteps) - num_inference_steps * self.scheduler.order
+
         with self.progress_bar(total=num_inference_steps) as progress_bar:
             for i, t in enumerate(timesteps):
+                if callback_on_step_end is not None:
```

Comment:
I think we won't be able to modify on the very last `latents` with this callback called on the beginning of next step, instead of end of current step - not sure if one ever need to modify the last `latents` though

Reply:
I think the final latents are exactly the same as the output?

Reply:
@TimothyAlexisVass 
yes! same as output but before decoding

Reply:
If someone really need to do something on the final latents it would be possible to use `output_type="latent"`.

Another way could be to keep the old callback and new modifying callback as unique.
That way one could still do the same type of non-modifying callbacks as before but use the new one for those that do.

Reply:
The callback should be moved to the end of the inner loop code - it's called "on step end" - this is pretty confusing as it's applied at step=1 before anything happens

Reply:
ok I will move to the end of inner loop code 

we will not be able to modify the inputs for the first step then, i.e. prompt_embeds, guidance_scale etc; but I think it's fine I think because we can pass the desired initial values to the input arguments ....